include/class.thread.php

```diff
@@ -993,14 +993,19 @@ function lookupByEmailHeaders(&$mailinfo, &$seen=false) {
                 //Lookup by ticket number
                 && ($ticket = Ticket::lookupByNumber($match[1]))
                 //Lookup the user using the email address
-                && ($user = User::lookup(array('emails__address' => $mailinfo['email'])))) {
+                && ($user = User::lookup(array('emails__address' => $mailinfo['email'])))
+                && ($staff = Staff::getIdByEmail($mailinfo['email']))) {
```

Comment:
I just re-commited.  The new commit accounts for both scenarios (when the staff might have a user ID in the system and when they do not)